.github/workflows/integration-tests.yml

```diff
@@ -26,6 +26,27 @@ jobs:
       - name: Run integration tests
         run: ./tests/integration-tests.sh
         shell: bash
+
+  test-api-key:
```

Comment:
We should be able to run these tests together with the integration tests, to save CI time on setting up another agent with the same characteristics

Reply:
I've set it up so that it runs after the other integration tests. It would need the services to run with different config so I couldn't put it as part of the other integration tests.